crates/gitbutler-project/src/controller.rs

```diff
@@ -120,16 +120,22 @@ impl Controller {
     }
 
     pub fn get(&self, id: ProjectId) -> Result<Project> {
-        self.get_inner(id, false)
+        self.get_inner(id, false, false)
+    }
+
+    /// Like [`Self::get()`], but will skip the git repository validation and directory creation.
+    /// This is intended to be used only when updating the path of a missing project.
+    pub fn get_raw(&self, id: ProjectId) -> Result<Project> {
+        self.get_inner(id, false, true)
     }
 
     /// Like [`Self::get()`], but will assure the project still exists and is valid by
     /// opening a git repository. This should only be done for critical points in time.
     pub fn get_validated(&self, id: ProjectId) -> Result<Project> {
-        self.get_inner(id, true)
+        self.get_inner(id, true, false)
     }
 
-    fn get_inner(&self, id: ProjectId, validate: bool) -> Result<Project> {
+    fn get_inner(&self, id: ProjectId, validate: bool, skip_dir_creation: bool) -> Result<Project> {
```

Comment:
Also, I'm not sure whether or not adding a third way of getting the project is the best idea here.

Maybe we can just skip the creation of the directory if `validate` is set to false, just wasn't sure if there are some cases in which we need to create the project directory even if there is no validation done